hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieDataFile.java

```diff
@@ -18,53 +18,53 @@
 
 import com.uber.hoodie.common.util.FSUtils;
 import java.io.Serializable;
-import java.util.Comparator;
 import org.apache.hadoop.fs.FileStatus;
 
 public class HoodieDataFile implements Serializable {
 
-  private FileStatus fileStatus;
+  // FileStatus not serializable in HDP 2.x (See: https://issues.apache.org/jira/browse/HADOOP-13895)
+  private transient FileStatus fileStatus;
+  private final String fileName;
+  private final String fullPathStr;
+  private final long len;
 
   public HoodieDataFile(FileStatus fileStatus) {
     this.fileStatus = fileStatus;
+    this.fileName = fileStatus == null ? null : fileStatus.getPath().getName();
```

Comment:
To check if a file status exists for a HoodieDataFile, we will have to check against each of these fields == null, should we add a method hasFileStatus() or something where you can perform this null check internally and then provide boolean to the user of this class ?